plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteAnd.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import static io.trino.spi.expression.StandardFunctions.AND_FUNCTION_NAME;
+
+public class RewriteAnd
```

Comment:
Instead of a class we could have a factory method on `RewriteLogicalExpression`

Reply:
i made `RewriteLogicalExpression` an impl detail of these classes (hence it's package private)